semgrep/semgrep/dependency_aware_rule.py

```diff
@@ -0,0 +1,108 @@
+from pathlib import Path
+from typing import Dict
+from typing import Generator
+from typing import List
+from typing import Tuple
+
+from dependencyparser.models import PackageManagers
+from dependencyparser.package_restrictions import dependencies_range_match_any
+from dependencyparser.package_restrictions import find_and_parse_lockfiles
+from dependencyparser.package_restrictions import ProjectDependsOnEntry
+
+from semgrep.error import SemgrepError
+from semgrep.rule import Rule
+from semgrep.rule_match import CoreLocation
+from semgrep.rule_match import RuleMatch
+from semgrep.verbose_logging import getLogger
+
+logger = getLogger(__file__)
+
+
+PACKAGE_MANAGER_MAP = {
+    "pypi": PackageManagers.PYPI,
+    "npm": PackageManagers.NPM,
+}
+
+
+def parse_depends_on_restrictions(
+    entries: List[List[Dict[str, str]]]
+) -> Generator[ProjectDependsOnEntry, None, None]:
+    for wrapper_entry in entries:
+        for entry in wrapper_entry:
+            # schema checks should gaurantee we have these fields, but we'll code defensively
+            namespace = entry.get("namespace")
+            if namespace is None:
+                raise SemgrepError(f"project-depends-on is missing `namespace`")
+            pm = PACKAGE_MANAGER_MAP.get(namespace)
+            if pm is None:
+                raise SemgrepError(
+                    f"unknown package namespace: {namespace}, only {list(PACKAGE_MANAGER_MAP.keys())} are supported"
+                )
+            package_name = entry.get("package")
+            if package_name is None:
+                raise SemgrepError(f"project-depends-on is missing `package`")
```

Comment:
Maybe worth making `project-depends-on` a constant since we're starting with `r2c-internal-only-project-depends-on`?

Reply:
I think it's ok for now; there's not an existing source of other constants to put it with, since this is destined for the core eventually anyways